packages/firebase/app/index.d.ts

```diff
@@ -970,6 +970,14 @@ declare namespace firebase.firestore {
     collection(collectionPath: string): CollectionReference;
 
     /**
+     * Returns true if this `DocumentReference` is equal to the provided one.
+     *
+     * @param other The `DocumentReference` to compare against.
+     * @return true if this `DocumentReference` is equal to the provided one.
+     */
+    isEqual(other: DocumentReference): boolean;
```

Comment:
Maybe this is gas on a fire, but why isEqual? Within our codebase I see usages like these:
equals 187
isEqual 53

Looking more closely, the 53 usages of isEqual are just database's Query.isEqual and its tests (some some checked in logs and Firestore's Query.isEqual).

It seems far more conventional to call this method `equals`.

(BTW: I'm finding usages like this)

```shell
find . \( -name node_modules \
    -o -name dist \
    -o -name \*.js \
    -o -name \*.map \) -prune -o -type f -print0 \
  | xargs -0 grep isEqual
```

Reply:
Okay, discussed offline, and it's clear that there's no consensus in the world about `isEqual` vs `equal`. The much larger incidence of equals within our code are internal APIs.

Let's file a clean up bug to standardize on `isEqual` so that `customDeepEqual` need not check both.